cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/guice/SourceControlModule.java

```diff
@@ -14,21 +14,33 @@
  * the License.
  */
 
-package io.cdap.cdap.sourcecontrol;
+package io.cdap.cdap.sourcecontrol.guice;
 
 import com.google.inject.AbstractModule;
 import com.google.inject.Scopes;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
 import io.cdap.cdap.sourcecontrol.operationrunner.InMemorySourceControlOperationRunner;
+import io.cdap.cdap.sourcecontrol.operationrunner.RemoteSourceControlOperationRunner;
 import io.cdap.cdap.sourcecontrol.operationrunner.SourceControlOperationRunner;
 
 /**
  *  Guice module for source control management classes.
  */
 public class SourceControlModule extends AbstractModule {
 
+  private final CConfiguration cConf;
+
+  public SourceControlModule(CConfiguration cConf) {
+    this.cConf = cConf;
+  }
   @Override
   protected void configure() {
-    // TODO: CDAP-20322 add the remoteSourceControlOperationRunner
-    bind(SourceControlOperationRunner.class).to(InMemorySourceControlOperationRunner.class).in(Scopes.SINGLETON);
+    boolean workerPoolEnabled = cConf.getBoolean(Constants.TaskWorker.POOL_ENABLE);
```

Comment:
Should we use the factory pattern or this conditional binding is sufficient? @chtyim @albertshau

Reply:
+1 I think that's the existing pattern too.

Reply:
Updated to Provider pattern